pkg/controller/sdk/reconciler.go

```diff
@@ -104,34 +101,49 @@ func (r *Reconciler) InjectConfig(c *rest.Config) error {
 	return err
 }
 
-func (r *Reconciler) statusHasChanged(ctx context.Context, old, new runtime.Object) (bool, error) {
+func (r *Reconciler) needsUpdate(ctx context.Context, old, new runtime.Object) bool {
 	if old == nil {
-		return true, nil
+		return true
 	}
 
-	o := NewReflectedStatusAccessor(old)
-	n := NewReflectedStatusAccessor(new)
+	// Check Status.
+	os := NewReflectedStatusAccessor(old)
+	ns := NewReflectedStatusAccessor(new)
+	oStatus := os.GetStatus()
+	nStatus := ns.GetStatus()
+
+	if !equality.Semantic.DeepEqual(oStatus, nStatus) {
+		return true
+	}
 
-	oStatus := o.GetStatus()
-	nStatus := n.GetStatus()
+	// Check finalizers.
+	of := NewReflectedFinalizersAccessor(old)
+	nf := NewReflectedFinalizersAccessor(new)
+	oFinalizers := of.GetFinalizers()
+	nFinalizers := nf.GetFinalizers()
 
-	if equality.Semantic.DeepEqual(oStatus, nStatus) {
-		return false, nil
+	if !equality.Semantic.DeepEqual(oFinalizers, nFinalizers) {
+		return true
 	}
 
-	return true, nil
+	return false
 }
 
-func (r *Reconciler) updateStatus(ctx context.Context, request reconcile.Request, object runtime.Object) (runtime.Object, error) {
+func (r *Reconciler) update(ctx context.Context, request reconcile.Request, object runtime.Object) (runtime.Object, error) {
```

Comment:
I would revert this and keep `updateStatus` and make a new `updateFinalizers` and `update`, have `update` call both those new methods. It will make testing easier.

Reply:
I want there to be only a single call to `r.client.Update()`. Do you think it is worth pulling the 4 lines updating status and 4 lines updating finalizers into their own methods?

I disagree about making testing easier, because (where possible) we should be testing only the public interface, not the internal methods directly. And it is possible to test this without needing direct access to update.

Reply:
ok